### ISSUE TITLE
pragmatic profiling: find out what it takes to fetch many BenchmarkResult objects for building an HTTP response

### ISSUE THREAD

**Issue by jgehrcke** [conbench/conbench#1035]
It becomes apparent that the individual `BenchmarkResult` is the core entity, and many UI views and API responses (HTTP responses in general) are built by processing potentially many of those.

Do some (rough) measurements to see where we are today in terms of timing, allowing for some back-of-the envelope calculations.

**Comment by jgehrcke** [comment]
An interesting data point comes from using the Arrow Conbench DB state in my local dev environment (strongish CPU):

- 164298 benchmark results distributed across 100 runs.
- In that case, 657192 calls to `sqlalchemy/orm/loading.py:1028(_instance)` took `3.6 s` wall time.
- and `1.6 s` of those `3.6 s` were spent alone in `json.loads()` (stdlib) with 492895 invocations.

I will look into documenting how I got those numbers.

**Comment by mariusvniekerk** [comment]
may be worth testing swopping the json implementation with cysimdjson or msgspec

**Comment by jgehrcke** [comment]
> may be worth testing swopping the json implementation with cysimdjson or msgspec

Right. For _serializing_ into JSON we have the related ticket https://github.com/conbench/conbench/issues/999 and started using orjson already: https://github.com/conbench/conbench/pull/1002.

For _deserealizing_ JSON we might want to use https://github.com/sqlalchemy/sqlalchemy/blob/6f3c741e41e13624d63f4faa4bdcec8466bda7f4/lib/sqlalchemy/dialects/postgresql/json.py#L198 and then maybe pick a library based on simdjson -- yes. But we will have to see, maybe orjson is also the right thing to use there. I found this rather insightful (emphasis mine):

> pysimdjson / cysimdjson are by far the fastest if you only need to parse documents and access **a few individual keys** (> 2x faster than orjson). As soon as you convert the simdjson result to a dict or iterate over all keys, orjson is the faster option. simdjson.dumps() basically does that, since it's only an alias to json.dumps(), so it's no surprise that orjson is faster in this benchmark. simdjson is primarily a parser, not a serializer, so measuring its serialization speed is not a good performance comparison.

from https://gist.github.com/kgriffs/e34364896ae2f329aa8016ec6f71fdce?permalink_comment_id=4177378#gistcomment-4177378

Here, we see that parsing with orjson is a really good choice:
https://github.com/tktech/json_benchmark#complete-load-of-datatwitterjson
https://github.com/tktech/json_benchmark#complete-load-of-datacanadajson

**Comment by jgehrcke** [comment]
> For deserealizing JSON we might want to use https://github.com/sqlalchemy/sqlalchemy/blob/6f3c741e41e13624d63f4faa4bdcec8466bda7f4/lib/sqlalchemy/dialects/postgresql/json.py#L198 and then maybe pick a library based on simdjson -- yes. But we will have to see, maybe orjson is also the right thing to use there.

doing that here: https://github.com/conbench/conbench/issues/1035

Btw @mariusvniekerk I also quickly tried cysimdjson -- in this case, it didn't add a noticeable gain over orjson.
